src/dist/dist.rs

```diff
@@ -717,11 +727,48 @@ fn try_update_from_dist_<'a>(
                 None => Vec::new(),
             };
 
-            let changes = Changes {
+            let mut changes = Changes {
                 explicit_add_components: profile_components,
                 remove_components: Vec::new(),
             };
 
+            let mut not_found_components = Vec::new();
+
+            for component in components {
+                if let Some(component) =
+                    m.find_targeted_component(component, &toolchain.target, &toolchain.target)?
+                {
+                    changes.explicit_add_components.push(component);
+                } else {
+                    use crate::dist::manifest::Component;
+                    not_found_components.push(Component::new(component.to_string(), None, false));
+                }
+            }
+
+            for target in targets {
+                let triple = TargetTriple::new(target);
+                if let Some(component) =
+                    m.find_targeted_component("rust-std", &toolchain.target, &triple)?
+                {
+                    changes.explicit_add_components.push(component);
+                } else {
+                    use crate::dist::manifest::Component;
+                    not_found_components.push(Component::new(
+                        "rust-std".to_owned(),
+                        Some(triple.clone()),
+                        false,
+                    ));
+                }
+            }
+
+            if !not_found_components.is_empty() {
+                return Err(Error::from(ErrorKind::RequestedComponentsUnavailable(
```

Comment:
I think this will mean that if you misspell a component, it will keep walking back nightlies all the way to 2014. Not sure how we avoid this, except perhaps to re-add a cutoff point for how far back we'll search when doing a fresh install?

Reply:
So, when I tried with `-c foo` I got:

```
info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'
info: latest update on 2019-09-27, rust version 1.40.0-nightly (ddf43867a 2019-09-26)
info: skipping nightly which is missing installed component 'foo'
thread 'main' panicked at 'Malformed manifest date: ParseError(TooShort)', src/libcore/result.rs:1084:5
```
I think this is because the badly named component caused an early exit,  as such perhaps it'd be easiest to just break the loop if `fetched` is empty after an attempt?

Reply:
Having tried that, I think it'll be right, so I'll amend for that:

```
info: syncing channel updates for 'nightly-x86_64-unknown-linux-gnu'
info: latest update on 2019-09-27, rust version 1.40.0-nightly (ddf43867a 2019-09-26)
error: component 'foo' is unavailable for download for channel x86_64-unknown-linux-gnu
```

being the result.